### ISSUE TITLE
Uploading many assets fails

### ISSUE THREAD

**Issue by letmejustputthishere** [dfinity/sdk#3152]
I tried the following:

1. Create asset folder with ~8k files (~600MB)
2. run `dfx deploy` to deploy asset canister and upload assets to it

I expected to see this happen: Files upload seamlessly

Instead, this happened: At the end of the process, this happens
```
Committing batch.
Error: Failed while trying to deploy canisters.
Caused by: Failed while trying to deploy canisters.
  Failed while trying to install all canisters.
    Failed to install wasm module to canister 'cherries_assets'.
      Failed to store assets in canister 'cherries_assets'.
        Failed asset sync with canister ypbrq-kqaaa-aaaae-qaeza-cai.
          Failed to synchronize frontend canister with project assets.
            The Replica returned an error: code 5, message: "Canister ypbrq-kqaaa-aaaae-qaeza-cai exceeded the instruction limit for single message execution."
```

### Meta

`dfx --version`:
```
`0.14.0`
```

<details><summary>Further log output</summary>
<p>

```
<log output>
```

</p>
</details>

**Comment by letmejustputthishere** [comment]
To solve this i have to upload the assets step by step (move more and more files into my `assets` folder),.

**Comment by sesi200** [comment]
This should be fixed by #3114
